crates/ruff_cli/src/diagnostics.rs

```diff
@@ -554,58 +555,40 @@ pub(crate) fn lint_stdin(
 }
 
 #[derive(Debug)]
-struct LintSources {
-    /// The "type" of source code, e.g. `.py`, `.pyi`, `.ipynb`, etc.
-    source_type: PySourceType,
-    /// The "kind" of source, e.g. Python file, Jupyter Notebook, etc.
-    source_kind: SourceKind,
-}
-
-impl LintSources {
-    /// Extract the lint [`LintSources`] from the given file path.
-    fn try_from_path(path: &Path) -> Result<LintSources, SourceExtractionError> {
-        let source_type = PySourceType::from(path);
-
-        // Read the file from disk.
+struct LintSource(SourceKind);
```

Comment:
It'd be nice to remove this, but `SourceKind` is in `ruff` and the `try_from_path` method relies on a lot of things that are in `ruff_cli`.